lib/goodcity/compare.rb

```diff
@@ -143,7 +143,7 @@ def compare_orders
     def preload_goodcity_objects(klass)
       memo_key = "preload_#{klass.to_s.underscore}"
       memo = instance_variable_get("@#{memo_key}")
-      memo ||= begin
+      memo || begin
```

Comment:
Actually this is important. Keep the `||=` here. We are effectively saying, "If memo hasn't been set, compute it and assign the result to memo and return memo." If we don't assign the result to `memo` then it will always be `nil` and we'll always re-compute the value.